app/src/main/java/com/codepath/michfeng/songswiper/connectors/RecommendationService.java

```diff
@@ -100,7 +105,22 @@ private ArrayList<String> getIDs(ArrayList<Track> tracks) {
     // Returns the SpotifyUser's recommendations.
     public ArrayList<Track> get(final VolleyCallBack callBack) {
         String endpoint = "https://api.spotify.com/v1/recommendations";
-        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET,endpoint,null, response -> {
+
+        initSeeds();
+
+        // Manipulate seeds to fit Spotify requirements (size 5, String array).
+        String [] seed_artist = topArtists.toArray(new String[topArtists.size()]);
+        String [] seed_genres = topGenres.toArray(new String[topGenres.size()]);
+        String [] seed_tracks = topTracks.toArray(new String[topTracks.size()]);
+
+        Map<String, String []> params = new HashMap();
+        params.put("seed_artists", Arrays.copyOfRange(seed_artist,0,5));
```

Comment:
Does it work to directly store them? Otherwise you may have two copies.
```
params.put("seed_artists", topArtists.toArray(new String[0]));
```
I also believe `new String[list.size()]` was replaced with `new String[0]`, which is actually faster due to JVM optimizations. https://shipilev.net/blog/2016/arrays-wisdom-ancients/